### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Andrewmazyar**

### CODE REVIEW

File: src/main/java/web/cinema/dao/impl/ShoppingCardDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    public ShoppingCart getByUser(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Query query = session.createQuery("from ShoppingCart c "
+                    + "left join fetch c.tickets Ticket "
+                    + "left join fetch c.user User "
+                    + "where c.user = :user", ShoppingCart.class);
```
Comment (dmytro-hryhoruk) [inline src/main/java/web/cinema/dao/impl/ShoppingCardDaoImpl.java:51]: you already have user here. Do you need to fetch it one more time?

File: src/main/java/web/cinema/dao/impl/TicketDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public Ticket add(Ticket ticket) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(ticket);
+            session.flush();
```
Comment (dmytro-hryhoruk) [inline src/main/java/web/cinema/dao/impl/TicketDaoImpl.java:24]: do we need to use flush() in our case?

File: src/main/java/web/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        List<Ticket> tickets = shoppingCart.getTickets();
+        tickets.add(ticket);
```
Comment (dmytro-hryhoruk) [inline src/main/java/web/cinema/service/impl/ShoppingCartServiceImpl.java:29]: you will add ticket to shopping cart without id

File: src/main/java/web/cinema/dao/impl/ShoppingCardDaoImpl.java (hunk 1 above)
Comment (Andrewmazyar) [reply src/main/java/web/cinema/dao/impl/ShoppingCardDaoImpl.java:51]: i see this case on QnA with boroda and do the same

**Comment by Andrewmazyar** [comment]
> travis checks have failed

**Comment by Andrewmazyar** [comment]
> travis checks have failed

very fast response after commit)))

File: src/main/java/web/cinema/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -24,6 +25,10 @@ public User add(User user) {
 
     @Override
     public User findByEmail(String email) {
-        return userDao.findByEmail(email);
+        Optional<User> user = userDao.findByEmail(email);
+        if (user.isPresent()) {
+            return user.get();
+        }
+        return null;
```
Comment (dmytro-hryhoruk) [inline src/main/java/web/cinema/service/impl/UserServiceImpl.java:32]:
```suggestion
   return userDao.findByEmail(email).orElse(null);;
```

**Pull request merged**